src/aleph/http/client.clj

```diff
@@ -498,401 +957,157 @@
+                                        :logger logger
+                                        :non-tun-proxy? non-tun-proxy?
+                                        :pipeline pipeline
+                                        :pipeline-transform pipeline-transform
+                                        :raw-stream? raw-stream?
+                                        :remote-address remote-address
+                                        :response-buffer-size response-buffer-size
+                                        :ssl-context ssl-context
+                                        :ssl? ssl?)]
+                  (log/debug (str "HTTP protocol: " protocol))
+
+                  (let [version-specific-req-fn
+                        (cond (.equals ApplicationProtocolNames/HTTP_1_1 protocol)
+                              (setup-http1-client setup-opts)
+
+                              (.equals ApplicationProtocolNames/HTTP_2 protocol)
+                              (do
+                                (setup-http2-pipeline setup-opts)
+                                (http2-req-handler setup-opts))
+
+                              :else
+                              (do
+                                (let [msg (str "Unknown protocol: " protocol)
+                                      e (IllegalStateException. msg)]
+                                  (log/error e msg)
+                                  (netty/close ch)
+                                  (throw e))))]
+
+                    ;; Both Netty and Aleph are set up, unpause the pipeline
+                    (when (.get pipeline "pause-handler")
+                      (log/debug "Unpausing pipeline")
+                      (.remove pipeline "pause-handler"))
+
+                    (fn http-req-fn
+                      [req]
+                      (log/trace "http-req-fn fired")
+                      (let [t0 (System/nanoTime)
+                            ;; I suspect the below is an error for http1
+                            ;; since the shared handler might not match.
+                            ;; Should work for HTTP2, though
+                            raw-stream? (get req :raw-stream? raw-stream?)]
+
+                        (if (or (not (.isActive ch))
+                                (not (.isOpen ch)))
+                          (d/error-deferred
+                            (ex-info "Channel is inactive/closed."
+                                     {:req     req
+                                      :ch      ch
+                                      :open?   (.isOpen ch)
+                                      :active? (.isActive ch)}))
+
+                          ;; If :aleph/close is set in the req, closes the channel and
+                          ;; returns a deferred containing the result.
+                          (if (or (contains? req :aleph/close)
+                                  (contains? req ::close))
```

Comment:
If think that `::close` can be removed now, no?

Reply:
I thought I replied to this already...I left in `::close` on the off chance some user was manually using it.